modules/eks-managed-node-group/main.tf

```diff
@@ -295,7 +295,7 @@ resource "aws_eks_node_group" "this" {
   }
 
   dynamic "remote_access" {
-    for_each = var.remote_access
+    for_each = length(var.remote_access) > 0 ? [var.remote_access] : []
```

Comment:
just testing locally looks like we also need to change lines 300-301 to:

```hcl
	ec2_ssh_key               = try(remote_access.value.ec2_ssh_key, null)
	source_security_group_ids = try(remote_access.value.source_security_group_ids, [])
```

Reply:
Nice catch! I have applied your fix. Thanks!